enterprise/backend/src/metabase_enterprise/workspaces/sync.clj

```diff
@@ -0,0 +1,24 @@
+(ns metabase-enterprise.workspaces.sync
+  (:require
+   [metabase.driver :as driver]
+   [metabase.driver.util :as driver.u]
+   [metabase.sync.core :as sync]
+   [metabase.sync.sync-metadata.tables :as sync-tables]
+   [metabase.sync.util :as sync-util]))
+
+;; this ns will be probably moved
+
+;; will figure out whether that is a good idea soonish
+;; this may be tricky!!!
```

Comment:
The more I sit with it, the more I think cloning the table metadata as an inactive record, with no fields, is the better way to go.

Then we can skip even creating the new tables until execution - it'll be much more snappy for high latency dwh's with lots of transforms being checked out.

Reply:
Thinking through what we need to do to the table when cloning:

1. Set workspace id.
2. Set schema to workspace schema.
3. Clear timestamps (let them be recalculated by hook)
4. Set active to false.
5. Clear initial_sync_status.
6. Clear estimated_row_count.
7. Clear is_defective_duplicate.
8. Recalculate unique_table_helper.
9. Clear unique
10. Set show_in_getting_started to false

The stuff from 6 down is getting into weird coupling that isn't ideal. Probably we can ignore some, but yeah.

The biggest risk is around new columns that get added, but my feeling is that we can protect ourselves by having a workspace and some cloned entities in the datasets used by our tests (particularly e2e).